src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/Android/PopupRenderer.android.cs

```diff
@@ -165,30 +171,30 @@ void SetSize()
 						break;
 				}
 
-				if (Element.Content.WidthRequest > -1)
+				if (realContentWidth > -1)
 				{
-					var inputMeasuredWidth = Element.Content.WidthRequest > Element.Size.Width ?
-						(int)Element.Size.Width : (int)Element.Content.WidthRequest;
+					var inputMeasuredWidth = realContentWidth > realWidth ?
+						realWidth : realContentWidth;
 					container.Measure(inputMeasuredWidth, (int)MeasureSpecMode.Unspecified);
 					horizontalParams = container.MeasuredWidth;
 				}
 				else
 				{
-					container.Measure((int)Element.Size.Width, (int)MeasureSpecMode.Unspecified);
-					horizontalParams = container.MeasuredWidth > Element.Size.Width ?
-						(int)Element.Size.Width : container.MeasuredWidth;
+					container.Measure(realWidth, (int)MeasureSpecMode.Unspecified);
+					horizontalParams = container.MeasuredWidth > realWidth ?
+						realWidth : container.MeasuredWidth;
 				}
 
-				if (Element.Content.HeightRequest > -1)
+				if (realContentHeight > -1)
```

Comment:
Not sure if `realContentHeight ` will work fine as it was calculated based on `Element.Size.Width`, while here it was `Element.Content.HeightRequest`